controllers/promiserelease_controller_test.go

```diff
@@ -92,79 +94,105 @@ var _ = Describe("PromiseReleaseController", func() {
 
 			It("sets the promise release status to installed", func() {
 				err := fakeK8sClient.Get(context.TODO(), promiseReleaseNamespacedName, &promiseRelease)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(promiseRelease.Status.Installed).To(BeTrue())
-			})
-		})
-
-		When("the sourceRef type is unknown", func() {
-			It("errors", func() {
-				promiseRelease.Spec.SourceRef.Type = "unknown"
-
-				err := fakeK8sClient.Create(context.TODO(), &promiseRelease)
-				Expect(err).ToNot(HaveOccurred())
-
-				_, err = t.reconcileUntilCompletion(reconciler, &promiseRelease)
-
-				Expect(err).To(MatchError("unknown sourceRef type: unknown"))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(promiseRelease.Status.Status).To(Equal("Installed"))
+				Expect(promiseRelease.Status.Conditions).To(HaveLen(1))
+				Expect(promiseRelease.Status.Conditions[0].Type).To(Equal("Installed"))
+				Expect(promiseRelease.Status.Conditions[0].Status).To(Equal(metav1.ConditionTrue))
+				Expect(promiseRelease.Status.Conditions[0].Message).To(Equal("Installed successfully"))
+				Expect(promiseRelease.Status.Conditions[0].Reason).To(Equal("InstalledSuccessfully"))
 			})
 		})
 
 		When("the sourceRef type is http", func() {
 			When("the url fetcher errors", func() {
 				BeforeEach(func() {
 					fakeFetcher.FromURLReturns(nil, fmt.Errorf("can't do mate"))
-					err := fakeK8sClient.Create(context.TODO(), &promiseRelease)
-					Expect(err).ToNot(HaveOccurred())
+					err = fakeK8sClient.Create(context.TODO(), &promiseRelease)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = t.reconcileUntilCompletion(reconciler, &promiseRelease)
 				})
 
-				It("installs the promise from the URL", func() {
-					_, err := t.reconcileUntilCompletion(reconciler, &promiseRelease)
+				It("errors", func() {
 					Expect(err).To(MatchError("failed to fetch promise from url: can't do mate"))
 				})
+
+				It("updates the status", func() {
+					err := fakeK8sClient.Get(context.TODO(), promiseReleaseNamespacedName, &promiseRelease)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(promiseRelease.Status.Status).To(Equal("Error installing"))
```

Comment:
status could be a bit more descriptive then always error 🤔

Reply:
yeah, fair enough; i think `Error Installing` kinda annoys me more than if it was just `Error` lol but nw